### ISSUE TITLE
`rule-name`: `no-useless-spread` disallows using spread to copy an array

### ISSUE THREAD

**Issue by hackel** [sindresorhus/eslint-plugin-unicorn#2631]
### Description

I'm getting a lot of no-useless-spread errors that I don't believe are "useless".  They are used to make a copy of an array.  This is important in e.g. Vue to trigger reactivity changes.

### Examples

Currently:
```js
// ❌
const array = [...a];

// ✅
const array = a; // array === a, not making a copy
```
Desired:
```js
// ✅
const array = [...a]; // array !== a
```

**Comment by fisker** [comment]
I don't think we report such case, what the message says? Can provide more context?

**Comment by hackel** [comment]
I've searched my logs and can't find what was causing this.  Sorry!  I'll re-open if I see it again.
